whylog/rulesbase/__init__.py

```diff
@@ -1,6 +1,60 @@
+import yaml
+from abc import ABCMeta, abstractmethod
 
+class AbstractConfigAccess(object):
 
-class WhylogBase(object):
+        """docstring for AbstractConfigAccess"""
 
-    def __init__(self, **kwargs):
-        pass
+        __metaclass__ = ABCMeta
+
+        @abstractmethod
+        def get_parsers(self):
+                pass
+
+        @abstractmethod
+        def get_rule(self, effect_name):
+                pass
+        
+class YamlConfigAccess(AbstractConfigAccess):
+        
+        """docstring for YamlConfigAccess"""
+
+        def __init__(self, parsers_path, rules_path):
+                self._parsers_path = parsers_path
+                self._rules_path = rules_path
+                self._load_parsers()
+                self._load_rules()
+
+        def _load_config_from_file(self, path):
+                config_file = open(path, "r")
+                loaded_yaml = yaml.load_all(config_file)
+                config = []
+                for yaml_doc in loaded_yaml:
+                        config.append(yaml_doc)
+                config_file.close()
+                return config
+
+        def _load_parsers(self):
+                parsers = self._load_config_from_file(self._parsers_path)
+                parsers_dict = {}
+                for parser in parsers:
+                        parsers_dict[parser['name']] = parser
+                self._parsers = parsers_dict
+
+        def _load_rules(self):
+                rules = self._load_config_from_file(self._rules_path)
+                rules_dict = {}
```

Comment:
you can use `defaultdict(list)` here (`from collections import defaultdict`) so that you don't need to pay O(log(n)) twice to add a rule and so that the code is cleaner